File: Day 21 DSA (678)/Code.cpp
class Solution {
public:
    bool checkValidString(string s) {
        int c = 0; 
        int n = s.size(); 
        for (int i = 0; i < n; ++i) {
            if (s[i] != ')') {
                ++c;
            }
            else if (c > 0) {
                --c;
            }
            else {
                return false;
            }
        }
        c = 0; 
        for (int i = n - 1; i >= 0; --i) {
            if (s[i] != '(') {
                ++c;
            } 
            else if (c > 0) {
                --c;
            }
            else {
                return false;
            }
        }
        return true;
    }
};

File: Day 35 DSA (434)/code.py
class Solution(object):
    def countSegments(self, s):
        """
        :type s: str
        :rtype: int
        """
        cnt = 0
        segm = False
        for c in s:
            if c != ' ' and not segm:
                cnt += 1
                segm = True
            elif c == ' ':
                segm = False
        return cnt
